google/oauth2/_client.py

```diff
@@ -510,7 +515,7 @@ def refresh_grant(
     return _handle_refresh_grant_response(response_data, refresh_token)
 
 
-def lookup_trust_boundary(request, url, access_token):
+def lookup_trust_boundary(request, url, headers=None):
```

Comment:
Shouldn't this be _lookup_trust_boundary?

Also, should this method handle adding auth to the call, instead of that burden being handled by callers?